pkg/api/api.go

```diff
@@ -87,3 +129,37 @@ func HealthSummary(c echo.Context) error {
 	}
 	return c.JSON(http.StatusOK, apiResponse)
 }
+
+func getBestPartitioner(totalChecks int, rangeDuration time.Duration) time.Duration {
+	if totalChecks <= desiredNumOfCheckStatuses {
```

Comment:
> if I have a check that runs once per hour, then I can go at least 4d before needing to aggregate

@moshloop In this case you mentioned earlier, we would have still not performed aggregation because the total checks (96) would be less than the desired num of checks (100).

However, I have made improvements to handle cases where aggregation is not done even if the total number of checks are above 100 _(example: an hourly check for the duration of 6 days, 8th case in the unit test)_